pyopenms_viz/_matplotlib/core.py

```diff
@@ -580,12 +580,93 @@ class MATPLOTLIBMobilogramPlot(MATPLOTLIBChromatogramPlot, MobilogramPlot):
 @APPEND_PLOT_DOC
 class MATPLOTLIBSpectrumPlot(MATPLOTLIB_MSPlot, SpectrumPlot):
     """
-    Class for assembling a matplotlib spectrum plot
+    Class for assembling a matplotlib spectrum plot.
     """
 
-    pass
+    def _create_figure(self):
+        if self.width is None or self.height is None:
+            self.fig, self.ax = plt.subplots()
+        else:
+            self.fig, self.ax = plt.subplots(figsize=(self.width / 100, self.height / 100), dpi=100)
 
+        self.ax.set_title(self.title)
+        self.ax.set_xlabel(self.xlabel)
+        self.ax.set_ylabel(self.ylabel)
 
+        # Ensure `self.canvas` references the Axes, not the Figure
+        self.canvas = self.ax
+        return self.ax
+
```

Comment:
This method is likely not needed as there is already this functionality in the MATPLOTLIBPLOT class